bots/pr/src/test/java/org/openjdk/skara/bots/pr/CSRBotTests.java

```diff
@@ -20,60 +20,76 @@
-            csr.setProperty("resolution", JSON.object().put("name", "Approved"));
-            issue.addLink(Link.create(csr, "csr for").build());
+            var censusBuilder = credentials.getCensusBuilder()
+                    .addReviewer(reviewer.forge().currentUser().id())
+                    .addCommitter(author.forge().currentUser().id());
+            var prBot = PullRequestBot.newBuilder().repo(bot).issueProject(issueProject).censusRepo(censusBuilder.build()).enableCsr(true).build();
+            var csrIssueBot = new CSRIssueBot(issueProject, List.of(author), Map.of(bot.name(), prBot));
+
+            // Run issue bot once to initialize lastUpdatedAt
+            TestBotRunner.runPeriodicItems(csrIssueBot);
 
-            var bot = new CSRPullRequestBot(repo, issues);
+            var csr = issueProject.createIssue("This is an approved CSR", List.of(), Map.of());
+            csr.setState(Issue.State.OPEN);
+            csr.setProperty("issuetype", JSON.of("CSR"));
+            issue.addLink(Link.create(csr, "csr for").build());
 
             // Populate the projects repository
             var localRepoFolder = tempFolder.path().resolve("localrepo");
-            var localRepo = CheckableRepository.init(localRepoFolder, repo.repositoryType());
+            var localRepo = CheckableRepository.init(localRepoFolder, author.repositoryType());
             var masterHash = localRepo.resolve("master").orElseThrow();
             assertFalse(CheckableRepository.hasBeenEdited(localRepo));
-            localRepo.push(masterHash, repo.authenticatedUrl(), "master", true);
+            localRepo.push(masterHash, author.authenticatedUrl(), "master", true);
 
             // Make a change with a corresponding PR
             var editHash = CheckableRepository.appendAndCommit(localRepo);
-            localRepo.push(editHash, repo.authenticatedUrl(), "edit", true);
-            var pr = credentials.createPullRequest(repo, "master", "edit", issue.id() + ": This is an issue");
+            localRepo.push(editHash, author.authenticatedUrl(), "edit", true);
+            var pr = credentials.createPullRequest(author, "master", "edit", issue.id() + ": This is an issue");
+            PullRequestUtils.postPullRequestLinkComment(issue, pr);
 
-            // Add CSR label
-            pr.addLabel("csr");
+            // Use CSRIssueBot to add CSR label
+            TestBotRunner.runPeriodicItems(csrIssueBot);
+            assertTrue(pr.store().labelNames().contains("csr"));
 
-            // Run bot
-            TestBotRunner.runPeriodicItems(bot);
+            // Approve CSR issue
+            csr.setState(Issue.State.CLOSED);
+            csr.setProperty("resolution", JSON.object().put("name", "Approved"));
 
+            // Run bot
+            TestBotRunner.runPeriodicItems(csrIssueBot);
             // The bot should have removed the CSR label
             assertFalse(pr.store().labelNames().contains("csr"));
```

Comment:
At this point, the label is removed, but the PR body still doesn't have the checkbox checked. I think the logic for the checkbox and the label need to be unified and consistent.

Reply:
I think the checkbox should be checked.

Reply:
Please see the latest commit. I have updated this test.

Reply:
Ah, checked again with latest and you are right. I may have looked at the wrong `body` variable as I was in a hurry.